### ISSUE TITLE
Change default ordering of terms in ribbon table

### ISSUE THREAD

**Issue by selewis** [geneontology/ribbon#58]
Currently the ordering is alphabetical (case insensitive)

Want to change it to follow the ordering of the ribbon/slim itself (and alphabetical within the list for that slim heading)

**Comment by lpalbou** [comment]
With this PR (https://github.com/geneontology/ribbon/pull/65), I ordered the terms for each aspect. It should be easier to browse them I think

<img width="1277" alt="screen shot 2018-09-26 at 4 17 07 pm" src="https://user-images.githubusercontent.com/24249870/46114473-ad89f980-c1a7-11e8-81fa-2affbe76522a.png">

However, this was a temporary manual fix, so we can let that ticket opened for when we'll be fetching the mapped terms directly from BioLink / Ontobio without the agr.js file

**Comment by hattrill** [comment]
FB ordering of slim terms in ribbon (if that helps)
![screen shot 2018-10-02 at 16 14 31](https://user-images.githubusercontent.com/11865746/46358300-a149e480-c65e-11e8-8d09-59f6e8f10b04.png)

**Comment by srengel** [comment]
curators agree that alphabetical ordering of terms in the ribbon is suboptimal and would instead prefer a 'biological' ordering for the display of terms.

action item from Alliance Biological Function meeting of Tu 10/2/18 was to come up with a proposed order, which i have done and am deposited here for comment (pasting list below, but also uploading a txt file in case that's easier for anyone).

i tried to stay as faithful to the original ordering we had on the Alliance site, which was very close to the example of the FB ordering of slim terms Helen pasted in this ticket yesterday.

i think i now realize why agr.js was necessary in the first place (instead of just pulling directly from the goslim_agr subset in the ontology file): to specify display order and alternative labels when the term_name is less than desirable.

namespace: molecular_function

id: GO:0016491
name: oxidoreductase activity
label_as: oxidoreductase

id: GO:0016787
name: hydrolase activity
label_as: hydrolase

id: GO:0016740
name: transferase activity
label_as: transferase

id: GO:0016874
name: ligase activity
label_as: ligase

id: GO:0030234
name: enzyme regulator activity
label_as: enzyme regulator

id: GO:0038023
name: signaling receptor activity
label_as: signaling receptor

id: GO:0005102
name: signaling receptor binding

id: GO:0005215
name: transporter activity
label_as: transporter

id: GO:0005198
name: structural molecule activity
label_as: structural molecule

id: GO:0008092
name: cytoskeletal protein binding

id: GO:0003677
name: DNA binding

id: GO:0003723
name: RNA binding

id: GO:0003700
name: DNA-binding transcription factor activity
label_as: DNA-binding transcription factor 

id: GO:0008134
name: transcription factor binding

id: GO:0036094
name: small molecule binding

id: GO:0046872
name: metal ion binding

id: GO:0030246
name: carbohydrate binding

id: GO:0097367
name: carbohydrate derivative binding

id: GO:0008289
name: lipid binding

id: GO:0003674
name: molecular_function
label_as: other functions

=============================

namespace: biological_process

id: GO:0007049
name: cell cycle

id: GO:0016043
name: cellular component organization

id: GO:0051234
name: establishment of localization

id: GO:0008283
name: cell proliferation

id: GO:0030154
name: cell differentiation

id: GO:0008219
name: cell death

id: GO:0032502
name: developmental process
label_as: development

id: GO:0000003
name: reproduction

id: GO:0002376
name: immune system process
label_as: immune system

id: GO:0050877
name: nervous system process
label_as: nervous system

id: GO:0050896
name: response to stimulus

id: GO:0023052
name: signaling

id: GO:0006259
name: DNA metabolic process
label_as: DNA metabolism

id: GO:0016070
name: RNA metabolic process
label_as: RNA metabolism

id: GO:0019538
name: protein metabolic process
label_as: protein metabolism

id: GO:0005975
name: carbohydrate metabolic process
label_as: carbohydrate metabolism

id: GO:1901135
name: carbohydrate derivative metabolic process
label_as: carbohydrate derivative metabolism

id: GO:0006629
name: lipid metabolic process
label_as: lipid metabolism

id: GO:0042592
name: homeostatic process
label_as: homeostasis

id: GO:0009056
name: catabolic process
label_as: catabolism

id: GO:0065009
name: regulation of molecular function

id: GO:0050789
name: regulation of biological process

id: GO:0007610
name: behavior

id: GO:0008150
name: biological_process
label_as: other processess

=============================
namespace: cellular_component

id: GO:0005576
name: extracellular region

id: GO:0005886
name: plasma membrane

id: GO:0005856
name: cytoskeleton

id: GO:0005829
name: cytosol

id: GO:0005739
name: mitochondrion

id: GO:0005794
name: Golgi apparatus

id: GO:0005783
name: endoplasmic reticulum

id: GO:0005634
name: nucleus

id: GO:0005694
name: chromosome

id: GO:0031410
name: cytoplasmic vesicle

id: GO:0005768
name: endosome

id: GO:0005773
name: vacuole

id: GO:0045202
name: synapse

id: GO:0030054
name: cell junction

id: GO:0042995
name: cell projection

id: GO:0032991
name: protein-containing complex

id: GO:0005575
name: cellular_component
label_as: other components


[GOribbonTermOrder100218.txt](https://github.com/geneontology/ribbon/files/2443719/GOribbonTermOrder100218.txt)

**Comment by srengel** [comment]
oops cellular_component is currently labeled on the Alliance site as 'other locations'.  i listed 'other components' above, but i didn't mean to specify a different label than what is currently used!  i'm fine with 'other locations' for CC root (id: GO:0005575 name: cellular_component).

**Comment by hattrill** [comment]
Hi Stacia, MF and CC look good to me. I had a small play around with the order of the BP ribbon, grouping it by broad themes: cell number, cell organisation, differentiation/dev, systems (I put behaviour  between nervous system and response to stimulus because they are often linked in a sensory system), and the rest as was.

cell cycle
cell proliferation
cell death
cellular component organization
establishment of localization
cell differentiation
development
reproduction
immune system
nervous system
behavior
response to stimulus
signaling
DNA metabolism
RNA metabolism
protein metabolism
carbohydrate metabolism
carbohydrate derivative metabolism
lipid metabolism
catabolism
homeostasis
regulation of molecular function
regulation of biological process
other processess

**Comment by vanaukenk** [comment]
Thanks @srengel 

MF looks good to me, as does Helen's version of BP.

I had a go at CC, trying to more closely group components related to the cell periphery and the endomembrane system:

id: GO:0005576 
name: extracellular region

id: GO:0005886 
name: plasma membrane

id: GO:0045202
 name: synapse

id: GO:0030054
 name: cell junction

id: GO:0042995
 name: cell projection

id: GO:0031410
 name: cytoplasmic vesicle

id: GO:0005768
 name: endosome

id: GO:0005773
 name: vacuole

id: GO:0005794 
name: Golgi apparatus

id: GO:0005783
 name: endoplasmic reticulum

id: GO:0005829
 name: cytosol

id: GO:0005739
 name: mitochondrion

id: GO:0005634 
name: nucleus

id: GO:0005694
 name: chromosome

id: GO:0005856
 name: cytoskeleton

id: GO:0032991
 name: protein-containing complex

id: GO:0005575
 name: cellular_component
 label_as: other components

**Comment by srengel** [comment]
i see that the Alliance GO ribbon term ordering is still alphabetical within aspect.  this is not correct.  Curators have specified 'Biological' ordering.  we should like the curator-specified ordering to be restored.

**Comment by lpalbou** [comment]
@srengel sorry for the delay.

The order has been fixed here: https://github.com/geneontology/ribbon/pull/71 and the new version of the ribbon package is published https://www.npmjs.com/package/@geneontology/ribbon, so the alliance can integrate this new version right away.

![screen shot 2018-10-26 at 11 52 46 am](https://user-images.githubusercontent.com/24249870/47586663-b2141000-d915-11e8-8fb1-86648d48033b.png)


**Notes**:
* I used MF and BP from Stacia and CC from Kimberly, let me know if you want something else
* yes the `agr.js` was created in the first place to allow specific ordering / relabeling of terms, but this ordering (and alternate labeling) should be stored in the GO ontology and provided directly by the API (e.g. each slim term could have an order index). As a general best practice, IMHO, the ribbon should only worries about showing the data, not modifying it
